leveldb_os/expiry_os_test.cc

```diff
@@ -1349,5 +1350,170 @@ TEST(ExpiryManifestTester, Compact1)
+    }   // OpenTestDB
+
+};  // ExpiryDBTester
+
+
+sExpiryDBObject SimpleData[]=
+{
+    {"aa", "one", 0},
+    {"bb", "two", 0},
+    {"cc", "three", 0},
+    {"dd", "four", 0},
+    {"ee", "five", 0}
+};
+
+
+/*
+ * Do simple writes, see if data disappears
+ *
+ */
+TEST(ExpiryDBTester, Simple)
+{
+    size_t obj_count, loop;
+    Status s;
+    sExpiryDBObject * cursor;
+    std::string buffer;
+    leveldb::Iterator * iterator;
+
+    // enable compaction expiry
+    m_Expiry->expiry_enabled=true;
+    m_Expiry->expiry_minutes=2;
+    m_Expiry->whole_file_expiry=false;
+
+    obj_count=sizeof(SimpleData) / sizeof(SimpleData[0]);
+
+    // load data (now in memory buffer)
+    for (loop=0, cursor=SimpleData; loop<obj_count; ++loop, ++cursor)
+    {
+        s=m_DB->Put(WriteOptions(), cursor->m_Key, cursor->m_Value);
+        ASSERT_OK(s);
+    }   // for
+
+    // verify we can find it
+    for (loop=0, cursor=SimpleData; loop<obj_count; ++loop, ++cursor)
+    {
+        s=m_DB->Get(ReadOptions(), cursor->m_Key, &buffer);
+        ASSERT_OK(s);
+    }   // for
+
+    // verify we can walk it
+    iterator=m_DB->NewIterator(ReadOptions());
+    for (loop=0, iterator->SeekToFirst(); loop<obj_count; ++loop, iterator->Next())
+    {
+        ASSERT_EQ(iterator->Valid(), true);
+    }   // for
+    ASSERT_EQ(iterator->Valid(), false);
+    delete iterator;
+    iterator=NULL;
+
+    // expiry set to 2 min, so shift 10
+    m_DB->ShiftClockMinutes(10);
+
+    // all data gone?
+    for (loop=0, cursor=SimpleData; loop<obj_count; ++loop, ++cursor)
+    {
+        s=m_DB->Get(ReadOptions(), cursor->m_Key, &buffer);
+        ASSERT_EQ(s.ok(), false);
+    }   // for
+
+    iterator=m_DB->NewIterator(ReadOptions());
+    iterator->SeekToFirst();
+    ASSERT_EQ(iterator->Valid(), false);
+    delete iterator;
+    iterator=NULL;
+
+    // force data from memory buffer to .sst file
+    //  (after shifting clock!!)
+    m_DB->SetClock(m_BaseTime);
+    m_DB->CompactRange(NULL, NULL);
+
+    // verify we can find it
+    for (loop=0, cursor=SimpleData; loop<obj_count; ++loop, ++cursor)
+    {
+        s=m_DB->Get(ReadOptions(), cursor->m_Key, &buffer);
+        ASSERT_OK(s);
+    }   // for
+
+    // verify we can walk it
+    iterator=m_DB->NewIterator(ReadOptions());
+    for (loop=0, iterator->SeekToFirst(); loop<obj_count; ++loop, iterator->Next())
+    {
+        ASSERT_EQ(iterator->Valid(), true);
+    }   // for
+    ASSERT_EQ(iterator->Valid(), false);
+    delete iterator;
+    iterator=NULL;
+
+    // expiry set to 2 min, so shift 10
+    m_DB->ShiftClockMinutes(10);
+
+    // all data gone?
+    for (loop=0, cursor=SimpleData; loop<obj_count; ++loop, ++cursor)
+    {
+        s=m_DB->Get(ReadOptions(), cursor->m_Key, &buffer);
+        ASSERT_EQ(s.ok(), false);
+    }   // for
+
+    iterator=m_DB->NewIterator(ReadOptions());
+    iterator->SeekToFirst();
+    ASSERT_EQ(iterator->Valid(), false);
+    delete iterator;
+    iterator=NULL;
+
```

Comment:
With the clock shifted forward 10 minutes, what happens if you compact again? Will it delete the records? Would be interesting to compact, shift the clock back to "now" and see if the records are gone.